cmake/options.cmake

```diff
@@ -0,0 +1,322 @@
+include(CMakeDependentOption)
+
+#------------------------------------------------------------------------------
+# Add an option with optional dependencies.
+#
+# Arguments:
+#   <DEFAULT_STATE> - Is the option enabled by default? (`ON` or `OFF`)
+#
+#   DEPENDS <expression>
+#     A list of expressions which must evaluate to true for the component to
+#     be made available. (Otherwise, the component will not be enabled, and the
+#     option for the component will be hidden.)
+#
+# Extra arguments are combined (with a single space) to form the description of
+# the option.
+#------------------------------------------------------------------------------
+function(drake_option NAME DEFAULT_STATE)
+  # "Fix" escaping so cmake_parse_arguments will split properly (note:
+  # cmake_parse_arguments splits twice, so escape two more levels).
+  string(REPLACE "\\" "\\\\\\" _args "${ARGN}")
+
```

Comment:
@bradking is this level of escaping needed with the old version of parse arguments?